src/Exception/OutOfSetException.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mathematicator\Numbers\Exception;
+
+/**
+ * Exception thrown when result cannot be provided because final number would be out of allowed set.
+ */
+class OutOfSetException extends NumberException
+{
+	public function __construct(string $haystack = '', ?string $allowed = null)
+	{
+		$allowed = $allowed ? " (allowed: $allowed)" : '';
```

Comment:
Please never use magic variable in string:

```suggestion
		$allowed = $allowed ? ' (allowed: ' . $allowed . ')' : '';
```

Reply:
It's not a magic variable, it's a string variable. Could you please clarify me why is the longer notation better in this case?

Reply:
Because variable is separated. Codestyle only.